.github/workflows/downstream_inclusion.yml

```
@@ -0,0 +1,148 @@
+---
+# This file is part of the NiAS project (https://github.com/nias-project).
+# Copyright NiAS developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+name: downstream inclusion
+
+on:
+  pull_request:
+    # defaults + "ready_for_review" (so full matrix is run when that's toggled)
+    types: [opened, synchronize, reopened, ready_for_review]
+
+concurrency:
+  group: downstream_inclusion_${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  inclusion_as_submodule:
+    strategy:
+      fail-fast: false
+    name: as submodule
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        shell: bash -elo pipefail {0}
+    timeout-minutes: 30
+    env:
+      TEST_PROJECT_DIR: submodule_test_project
+    steps:
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Setup uv
+      uses: yezz123/setup-uv@v4
```

```suggestion
    - name: Setup uv
      uses: astral-sh/setup-uv@v5
```
THere's an official action now

---

Does the cmake build use `uv` internally?

---

It does.

---

Thanks, switched the action to the official one. Yes, we use uv internally, but if it is not available, it should be downloaded automatically. So I removed the setup action for the non-virtualenv tests.